server/src/rooms/schema/MyRoomState.ts

```diff
@@ -1,13 +1,13 @@
-import { Schema, Context, type, MapSchema } from "@colyseus/schema";
-
-
-export class Player extends Schema {
-  @type("number") x: number;
-  @type("number") y: number;
-  @type("string") pos: string;
-  @type("string") lastMovedTime: string;
-  @type("boolean") isMoving: boolean;
-}
+import {
+  Schema,
+  Context,
+  type,
+  MapSchema,
+  ArraySchema,
+} from "@colyseus/schema";
+import { Monster, Player } from "./Character";
+import { BattleRoom } from "../BattleRoom";
+import { BattleTeam } from "./Group";
 
 export class MyRoomState extends Schema {
   @type({ map: Player }) players = new MapSchema<Player>();
```

Comment:
this is the map i am referring to that maps sessionId/playerId to player Object. I'm not entirely sure on this. But this should allow u to handle the deletion of player when they leave battle nicely.